core/templates/pages/exploration-player-page/new-lesson-player/new-lesson-player-components/player-header.component.ts

```diff
@@ -0,0 +1,176 @@
+import { ReadOnlyExplorationBackendApiService } from 'domain/exploration/read-only-exploration-backend-api.service';
+import { StoryPlaythrough } from 'domain/story_viewer/story-playthrough.model';
+import { LearnerExplorationSummaryBackendDict } from 'domain/summary/learner-exploration-summary.model';
+import { ReadOnlyTopic } from 'domain/topic_viewer/read-only-topic-object.factory';
+import { TopicViewerBackendApiService } from 'domain/topic_viewer/topic-viewer-backend-api.service';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { Subscription } from 'rxjs';
+import { ContextService } from 'services/context.service';
+import { UrlService } from 'services/contextual/url.service';
+import { I18nLanguageCodeService, TranslationKeyType } from 'services/i18n-language-code.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { StatsReportingService } from '../../services/stats-reporting.service';
+
+import './player-header.component.css';
+
+
+@Component({
+  selector: 'oppia-player-header',
+  templateUrl: './player-header.component.html',
+  styleUrls: ['./player-header.component.css']
+})
+export class PlayerHeaderComponent {
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion. For more information, see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  explorationId!: string;
+  explorationTitle!: string;
+  explorationTitleTranslationKey!: string;
+  storyPlaythroughObject!: StoryPlaythrough;
+  topicName!: string;
+  topicNameTranslationKey!: string;
+  isLinkedToTopic!: boolean;
+  expInfo!: LearnerExplorationSummaryBackendDict;
+  directiveSubscriptions: Subscription = new Subscription();
+
+  constructor(
+    private contextService: ContextService,
+    private readOnlyExplorationBackendApiService:
+    ReadOnlyExplorationBackendApiService,
+    private siteAnalyticsService: SiteAnalyticsService,
+    private statsReportingService: StatsReportingService,
+    private urlInterpolationService: UrlInterpolationService,
+    private urlService: UrlService,
+    private i18nLanguageCodeService: I18nLanguageCodeService,
+    private topicViewerBackendApiService: TopicViewerBackendApiService
+  ) {}
+
+  ngOnInit(): void {
+    let pathnameArray = this.urlService.getPathname().split('/');
+    let explorationContext = false;
+
+    for (let i = 0; i < pathnameArray.length; i++) {
+      if (pathnameArray[i] === 'explore' ||
+          pathnameArray[i] === 'create' ||
+          pathnameArray[i] === 'skill_editor' ||
+          pathnameArray[i] === 'embed' ||
+          pathnameArray[i] === 'lesson') {
+        explorationContext = true;
+        break;
+      }
+    }
+
+    this.explorationId = explorationContext ?
+      this.contextService.getExplorationId() : 'test_id';
+
+    this.explorationTitle = 'Loading...';
```

Comment:
Why exploration title is initialized with 'Loading...'?

Reply:
So that the title isn't blank until we get a response from the API, also that isn't my code but the current lesson player's component duplicated

Reply:
Hmm, makes sense